Code/Framework/AzCore/Tests/DOM/DomPrefixTreeBenchmarks.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Tests/DOM/DomFixtures.h>
+#include <AzCore/DOM/DomPrefixTree.h>
+
+#define REGISTER_TREE_BENCHMARK(BaseClass, Method) \
+    BENCHMARK_REGISTER_F(BaseClass, Method)->Args({10, 1})->Args({1000, 1})->Args({10, 5})->Args({1000, 5})
+
+namespace AZ::Dom::Benchmark
+{
+    class DomPrefixTreeBenchmark : public Tests::DomBenchmarkFixture
+    {
+    public:
+        void SetUpHarness() override
+        {
+            Tests::DomBenchmarkFixture::SetUpHarness();
+            m_registeredPaths = AZStd::make_unique<AZStd::vector<Path>>();
```

Comment:
Just curious. Why create a unique pointer around the vector? This could very well be a vector as well right?

Reply:
Allocators make that a bit tricky: this actually can't be just a vector because we need its memory deallocated before the allocators are torn down, so the test harness will actually assert if we use a regular vector.